include/s3select_oper.h

```diff
@@ -701,21 +769,41 @@ class value
   }
   bool operator<=(const value& v)
   {
-    return !(*this>v);
+    if ((is_null() || v.is_null()) || (is_nan() || v.is_nan())) {
+      return false;
+    } else {
+      return !(*this>v);
+    } 
   }
+  
   bool operator>=(const value& v)
   {
-    return !(*this<v);
+    if ((is_null() || v.is_null()) || (is_nan() || v.is_nan())) {
+      return false;
+    } else {
+      return !(*this<v);
+    } 
   }
+  
   bool operator!=(const value& v)
   {
-    return !(*this == v);
+    if ((is_null() || v.is_null()) || (is_nan() || v.is_nan())) {
+      return true;
+    } else {
+      return !(*this == v);
+      } 
   }
-
+  
   template<typename binop> //conversion rules for arithmetical binary operations
   value& compute(value& l, const value& r) //left should be this, it contain the result
   {
     binop __op;
+    if ((l.is_null() || r.is_null()) || (l.is_nan() || r.is_nan()))
```

Comment:
optimization issue.
it possible to validate null/nan condition,  upon other conditions (less rare) are false.
and save some instructions per binary-operation